### PR TITLE
feat(augmentation): Add min_shift parameter to FractionalShift #284

### LINKED ISSUE #284: Add `min_shift` to FractionalShift
**Motivation**

In a custom pipeline with Data Augmentation  I used a function similar to FractionalShift but with an additional `min_shift` parameter.  The reasoning behind this is that adding data that is too similar to the original un-augmented data does not meaningfully increase variance in the input. 

I intend to move my code from custom implementations to `chemotools` and noticed this missing parameter.

**Feature description**

Add a parameter `min_shift: float | None = None` to `FractionalShift` so that the shift values are drawn uniformely from 

$`[-\text{shift}, -\text{min\_shift}] \cup [\text{shift}, \text{min\_shift}]`$

This would require code like this [here](https://github.com/paucablop/chemotools/blob/main/chemotools/augmentation/_fractional_shift.py#L153C9-L153C59):

```python
if self.min_shift:
    mag = self._rng.uniform(self.min_shift, self.shift)
    sign = self._rng.choice([-1.0, 1.0])
    shift = sign * mag
else:
    shift = self._rng.uniform(-self.shift, self.shift)  # current code
```

This is a fairly minimal, downward-compatible change, requiring the code above plus new tests. I'd be happy to provide a PR if you want to add this parameter.

### PR DESCRIPTION
**Pull request by m-ad**
## Summary

Implements `min_shift` parameter for `FractionalShift` as described in #284. Includes tests.

## Closes

Closes #284

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Refactor
- [ ] Performance improvement
- [ ] Documentation update
- [ ] Tests only
- [ ] CI / DevOps / tooling
- [ ] Dependency update
- [ ] Breaking change

## What changed?

- Implemented `min_shift` parameter for `FractionalShift`
- Added tests
- 

## What did NOT change?

Changes are backwards-compatible (`min_shift` defaults to `None`).

## API and compatibility impact

- [ ] No public API changes
- [ ] Public API added
- [x] Public API changed
- [ ] Deprecation introduced
- [ ] Breaking change introduced

### Notes
- scikit-learn API compatibility:
- Affected modules/classes/functions: `test_fractional_shift.py`, `_fractional_shift.py`
- Backward compatibility considerations: backwards-compatible

## Validation

- [x] `task format-check`
- [x] `task lint`
- [x] `task spelling`
- [x] `task type-check`
- [x] `task test`
- [x] `task coverage`
- [ ] `task test:matrix`
- [x] `task docs:check`
- [x] `task build`

### Validation notes

- 

## Tests

- [x] Added new tests
- [ ] Updated existing tests
- [ ] No tests added because this PR only changes docs / CI / metadata

### Test coverage details
- Relevant test files: `test_fractional_shift.py`
- Edge cases covered: `min_shift > shift` raises `ValueError`
- Numerical / estimator behavior checked:

## Documentation

- [ ] Docs updated
- [x] Docstrings updated
- [ ] No docs update needed

### Documentation notes

- 

## Dependency / build / CI impact

- [x] No dependency changes
- [ ] Runtime dependency change
- [ ] Dev dependency change
- [ ] GitHub Actions / CI changed
- [ ] Build/release process changed

### Notes
- 

## Reviewer focus

Please focus on:
- 
- 

## Screenshots / logs / benchmark output

<details>
<summary>Details</summary>
</details>

```text
Paste logs, benchmark numbers, or screenshots description here
```

## Checklist
- [x] I linked the relevant issue(s) or explained why none exists
- [x] I kept this PR scoped to a single purpose
- [x] I added or updated tests where appropriate
- [x] I updated documentation where appropriate
- [x] I verified the change locally using the relevant tasks above
- [x] I considered backward compatibility and public API impact
- [x] I am ready for review

### CODE REVIEW

**Comment by paucablop** [comment]
Thanks a lot for the well structured PR @m-ad , I will look into it this weekend 😄 

By the way, I am planning a refactoring of the augmentation module #209 . I also have some ideas to add additional augmentation functions, I would love to hear if you have any suggestion 😄

File: tests/augmentation/test_fractional_shift.py [hunk 1]
```diff
@@ -116,3 +117,73 @@ def test_fractional_shift_linear():
     assert spectrum_left_shifted[0][4] == 4.623620356542087
     assert spectrum_right_shifted[0][-1] == 11.0
     assert spectrum_left_shifted[0][0] == 0
+
+
+def test_fractional_shift_min_shift():
```
Comment (paucablop) [inline tests/augmentation/test_fractional_shift.py:122]: Nice test!

**Review by paucablop — APPROVED** [review]
Looks good! thanks for the PR :)

**Comment by paucablop** [comment]
Python CI / compatibility with python 3.12 min `scikit-kelarn` version fails because specific versions are not pinned. This issue is not related to this PR but and hence it is addressed on a separate issue (#286)

File: chemotools/augmentation/_fractional_shift.py [hunk 2]
```diff
@@ -33,6 +33,11 @@ class FractionalShift(
         Maximum absolute shift applied to each signal.
         A random shift is drawn uniformly from [-shift, +shift].
 
+    min_shift : float or None, default=None
+        Minimum absolute shift applied to each signal. When provided, the
+        shift is drawn uniformly  from the union of [-shift, -min_shift]
+        and [min_shift, shift].
+
     padding_mode : {'zeros', 'constant', 'wrap', 'extend',
         'mirror', 'linear'}, default='linear'
```
[the remark this answers is not in the document]
Comment (paucablop) [reply chemotools/augmentation/_fractional_shift.py:42]: Handeled on #277
